### DISCUSSION TITLE
Not seeing images for labelling in interactive mode

### DISCUSSION THREAD

**Discussion by ElisonSherton** [jina-ai/finetuner#157] [Q&A] [answered]
Hi guys,

![image](https://user-images.githubusercontent.com/45599762/138263236-fe54a544-cced-48ac-b783-a4f01152d50e.png)

I was trying to use finetuner to build a CBIR system and followed this [starter](https://finetuner.jina.ai/get-started/celeba/) but was getting the above error.

Environment Details: python3.8 with torch and finetuner installed.

Would be grateful if someone can clarify this earnestly.

Thanks,
Vinayak.

**Comment by bwanglzu** [comment]
hi @ElisonSherton i just runned the same code in my mac, I can not reproduce, can you try to install the latest jina using `pip install -U jina` and run again? 

also, can you check you have converted datauri to blob in `data_gen` like this?

```python
def data_gen():
    for d in from_files('/Users/bo/Downloads/img_align_celeba/*.jpg', size=100, to_dataturi=True):
        d.convert_image_datauri_to_blob(color_axis=0) 
        yield d
```

and please make sure the data path you send to `from_files` has been replaced to your local data path

**Reply by ElisonSherton** [reply]
Hi @bwanglzu 

Thanks for the prompt reply.

```python
from jina.types.document.generators import from_files

def data_gen():
    for d in from_files('../data/cleaned_length/womens-length/train/knee/*.jpg', size=100, to_dataturi=True):
        d.convert_image_datauri_to_blob(color_axis=0)  # no need of tf
        yield d

import torchvision
model = torchvision.models.resnet50(pretrained=True)

import finetuner

finetuner.fit(
    model=model,
    interactive=True,
    train_data=data_gen,
    freeze=True,
    to_embedding_model=True,
    input_size=(3, 224, 224),
    output_dim=100
)
```

Here is my code and I am able to see the UI, can't see any neighbours though as shown in the image below.

![image](https://user-images.githubusercontent.com/45599762/138303627-3e934e79-6c61-4f23-9a91-a21f29a7b993.png)

The path exists and also the UI is picking query images perfectly from the path, however recommendations for labelling are not shown.

Thanks,
Vinayak.

**Reply by bwanglzu** [reply]
hi @ElisonSherton 

thanks for the reply, I think I know the issue. can you change the file path to **absolute path**, instead of a relative path? We already noticed this issue in core, and [we're fixing it](https://github.com/jina-ai/jina/pull/3722).

Best

Bo

**Reply by ElisonSherton** [reply]
Hi @bwanglzu 

![image](https://user-images.githubusercontent.com/45599762/138311565-aa860dda-507f-4778-a9ab-0f1daecfc8ad.png)

I have changed my path to absolute path, but still I can see the same error and other suggestions are not getting loaded like before :/

Thanks,
Vinayak

**Comment by hanxiao** [comment]
i think the problem is not about the path, it complains about the data size, so I think your image size is not something that ResNet appreciate. ResNet accepts only `input_size=(3, 224, 224),` and from your log it says `3,500,500` or `3,900,600`

**Reply by ElisonSherton** [reply]
Hi @hanxiao ,

I have specified `input size` to be cast to `(3, 224, 224)` in the `finetuner.fit` method. Would that not suffice?

Do I need to resize everything beforehand to this size and save the images on disk and then load from there?

Thanks,
Vinayak

**Comment by hanxiao** [comment]
aha, okay i see what you mean,  actually `input_size` is the shape of the input layer of your network, finetuner requires this if your network is built by pytorch/paddle for doing its magic. But this argument is not about auto reshaping your input into `input_size`.

> Do I need to resize everything beforehand to this size and save the images on disk and then load from there?

No need, let me try something and get back to you

**Comment by hanxiao** [comment]
hi @ElisonSherton i'm guessing your images are in different sizes right? the first image's size is different than the second, and third, etc.

**Reply by ElisonSherton** [reply]
Yes, they're not of the same size.

**Reply by hanxiao** [reply]
that's the problem, i can also reproduce it when the size is not the same. it is a missing feature on our end, you can also normalize them by yourself first, or wait for a day we fix on our end

**Reply by hanxiao** [reply]
@ElisonSherton solving this issue in upstream https://github.com/jina-ai/jina/pull/3750

**Reply by vahuja4** [reply]
@hanxiao - has this been resolved now?

**Reply by hanxiao** [reply]
yes, see next thread.

**Comment by hanxiao** [comment, marked as answer]
@vahuja4  this is solved, and here is how you can use images of different sizes.

1. I prepared some random jpgs
![image](https://user-images.githubusercontent.com/2041322/138658846-8438b96c-d80a-4b06-a6a6-2b2b683707f2.png)
1. upgrade your finetuner, `pip install -U finetuner`
2. build the constructor with proper resizing:
```python
from jina.types.document.generators import from_files

def data_gen():
    for d in from_files('/Users/hanxiao/Documents/tmpimg/*.jpg'):
        d.convert_image_uri_to_blob()
        d.convert_uri_to_datauri()
        d.resize_image_blob(224, 224)
        d.normalize_image_blob()
        d.set_image_blob_channel_axis(-1, 0)
        yield d
```
3. Now run the main routine of finetuner.
```python
import torchvision
import finetuner

model = torchvision.models.resnet50(pretrained=True)

finetuner.fit(
    model=model,
    interactive=True,
    train_data=data_gen,
    freeze=True,
    to_embedding_model=True,
    input_size=(3, 224, 224),
    output_dim=100,  # Chop-off the last fc layer and add a trainable linear layer.
)
```

you will get:
![image](https://user-images.githubusercontent.com/2041322/138658761-97b72e70-46d3-4f8a-8eae-7cabbf34b5d4.png)
